spec/API_specification/array_api/__init__.py

```diff
@@ -0,0 +1,14 @@
+from .array_object import *
+from .constants import *
+from .creation_functions import *
+from .data_type_functions import *
+from .data_types import *
```

Comment:
```suggestion
```
`data_types` just defines `__eq__` in it's all, rather than a dtype object unlike say `array_object.array`. Might want to make a dtype object, but IMO I think just remove this line is fine enough for now—I think we want to avoid having an `__eq__` method in the top-level namespace as it could be confused for `array.__eq__`.

Also could you change this in `data_types.py`:
```diff
- __all__ = [__eq__]
+ __all__ = ['__eq__']
```
i.e. `__all__` should cover the string names, not the objects/functions themselves.

Reply:
Good catch, thanks @honno!

Reply:
Hmmm I feel in this case we still want this:
```python
import data_types
```
so we can document `data_types.__eq__`. Not perfect but the docs rely on it...

Reply:
Oh yeah, maybe then

```python
import data_types as dtype
```

Reply:
Re: string name exports. Addressed in https://github.com/data-apis/array-api/pull/430/commits/d3df3fcc4761ab300827cb4d2adfb1fc7e277e96.

Re: `import data_types`. Addressed in https://github.com/data-apis/array-api/pull/430/commits/73f9651624b75cd40aba19e94158651bcf51b65b.